File: 01-arrays_&_hashing/00217-Contains_Duplicate/00217-contains_duplicate.java
// 217: Contains Duplicate
// https://leetcode.com/problems/contains-duplicate/

import java.util.Set;
import java.util.HashSet;

class Solution {
    // SOLUTION
    public boolean containsDuplicate(int[] nums) {
        Set<Integer> set = new HashSet<Integer>();
		
        for(var i : nums)
			if(!set.add(i))
				return true; 
		
        return false;
    }

    public static void main(String[] args) {
        Solution o = new Solution();
        
        // INPUT
        int nums[] = {1,2,3,1};

        // OUTPUT
        var result = o.containsDuplicate(nums);
        System.out.println((result ? "true" : "false"));
    }
}

File: 01-arrays_&_hashing/00217-Contains_Duplicate/00217-contains_duplicate.cpp
// 217: Contains Duplicate
// https://leetcode.com/problems/contains-duplicate/

#include<iostream>
#include<vector>
#include<unordered_set>

using namespace std;

class Solution {
public:
    // SOLUTION
    bool containsDuplicate(vector<int>& nums) {
        unordered_set<int> us;
        for (int i=0; i<nums.size(); i++)
            if (us.insert(nums[i]).second == false)
                return true;
        return false;
    }
};

int main() {
    Solution o;

    // INPUT
    vector<int> nums = {1,2,3,1};

    // OUTPUT
    auto result = o.containsDuplicate(nums);
    cout << (result ? "true" : "false") << endl;
    
    return 0;
}
